### DISCUSSION TITLE
[Q] Move from esphome EAWS to MC EAWS

### DISCUSSION THREAD

**Discussion by MoonModules** [mathieucarbou/ESPAsyncWebServer#121] [General]
Continuing discussion started here: https://github.com/esphome/ESPAsyncWebServer/pull/38

Hi Matthieu,

I upgraded to 3.3.10 

The good news is it compiles now, 
The bad news is dat web socket data is not exchanged with the browser yet ...

MC webserver shows this in my serial output
Webserver: server.on serveIndex csdata 1061180544-24558 (/)αcaptivePortal 210 207
WS Client disconnected client: 2 ip:0.0.0.0 q:0 l:0 s:1 (#:0)
WS client connected client: 3 ip:192.168.1.183 q:0 l:0 s:1 (#:0)

Where esphome version shows this
Webserver: server.on serveIndex csdata 1061180676-24558 (/)αcaptivePortal 210 207
WS client connected client: 1 ip:192.168.1.183 q:0 l:0 s:1 (#:1)

So for some reason It disconnects ... you have any idea why?

Code changes can be found here: https://github.com/MoonModules/StarLight/commit/daeda4146e3970a6469fe507ecef91d863cd4547

(which gives a good overview - for me at least - what needs to be changed in case of moving from esphome to your environment.)

So I also saw this is not compiling anymore:
wsBuf->lock(); and unlock
ws._cleanBuffers();

Is that not necessary anymore?

Quote 'Also maybe try to fix the warnings: there are so many 😅'
    normally I only have warnings from imorted libraries so this was 'wip' warnings ;-)

**Comment by mathieucarbou** [comment]
> Is that not necessary anymore?

No it is not: locking is done inside the lib and buffers are shared_ptr.

You are doing too many changes I think to really be able to pinpoint the source fo the disconnection. 
But I am pretty sure it is linked to how it is used, otherwise there would have been countless of reported issues already ;-)

**Comment by ewowi** [comment]
The changes where the minimum needed to make it compile …
But you are right, I will dive into it and sort it out myself

thx for your help 🙏

**Comment by mathieucarbou** [comment]
Let's move that to an open discussion that will remain :-)
